### PR TITLE
[Refactor] Templates normalization

### PR DESCRIPTION
**Pull request by JulesGuesnon**
# Description

The goal of this PR is to be the base for a big refactor of all templates. The goal will be to have:
- Same naming for directories and env variables in `.env`
- Having a more friendly logging (Basically logging spinner when there's a loading)
- Replacing all `.env` by `.env.example` to avoid having sensitive information being commited by user after they cloned the repo
- Being able to deploy only with a `PK` being required. It implies that all env vars must have a default value if they're required. The 2 main reasons for it are:
    - Easier for user to deploy an example
    - Easier to deploy template in the registry

### CODE REVIEW

**Comment by JulesGuesnon** [comment]
# Example of logging


https://user-images.githubusercontent.com/36401048/208478600-5706a372-6501-43f7-b143-f5e73dbed85b.mov




https://user-images.githubusercontent.com/36401048/208479171-cb148962-6709-4bff-a014-0fffb2b1b13b.mov

File: .gitignore [hunk 1]
```diff
@@ -6,3 +6,4 @@ _esy/
 **/compiled
 /deploy/metadata.json
 /deploy/deploy.js
+deploy/.env
```
Comment (Laucans) [inline .gitignore]: replace it by **/.env ?

**Review by Laucans — APPROVED** [review]

File: deploy/.env.example [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Required: Your private key
```
Comment (Laucans) [inline deploy/.env.example]: Maybe use the extension .dist instead

**Review by Laucans — CHANGES_REQUESTED** [review]

File: deploy/.env.example (hunk 2 above)
Comment (Laucans) [reply deploy/.env.example]: It's convention which is already adopted by some templates 

> .dist files are often configuration files which do not contain the real-world deploy-specific parameters (e.g. Database Passwords, etc.), and are there to help you get started with the application/framework faster.

**Pull request merged**